### PR TITLE
feat: add pre-commit and various hook files

### PR DESCRIPTION
**Pull request by MarkoSagadin**
# Description

This PR adds the pre-commit config file and other config files to the template. 

## Areas of interest for the reviewer

Added files. Do you see places where documentation in the config files could be improved?

Also, any styling opinions are welcome. I have set the most disputed setting (markdown text wrap, see .prettierrc) to my preference.
 

## After-review steps

- I will merge PR by myself.

[style guidelines]:
  https://github.com/IRNAS/irnas-guidelines-docs/blob/main/docs/developer_guidelines.md

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
* **#23** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/23?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/21?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#20** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/20?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MarkoSagadin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "overrides": [
+        {
+            "files": "**/*.md",
+            "options": {
+                "proseWrap": "always",
+                "printWidth": 80
```
Comment (TjazVracko) [inline .prettierrc:7]:
```suggestion
                "printWidth": 100
```

🥺?

File: committed.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+subject_length = 72
+style = "conventional"
+subject_capitalized=false
+allowed_types = ["feat", "fix", "docs", "refactor", "style", "test", "infra", "chore"]
```
Comment (TjazVracko) [inline committed.toml:4]: Maybe "chore" should be "misc"?

**Review by TjazVracko — APPROVED** [review]
LGTM,

I have made 2 suggestions, no need to apply them :)

File: .prettierrc (hunk 1 above)
Comment (MarkoSagadin) [reply .prettierrc:7]: I have increased width to 100 :)

File: committed.toml (hunk 2 above)
Comment (MarkoSagadin) [reply committed.toml:4]: Changed to misc.

**Comment by MarkoSagadin** [comment]
### Merge activity

* **Mar 28, 10:53 AM EDT**: @MarkoSagadin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/20).

**Pull request merged**